presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveUtil.java

```diff
@@ -256,7 +256,8 @@ private HiveUtil()
             RecordReader<WritableComparable, Writable> recordReader = (RecordReader<WritableComparable, Writable>) inputFormat.getRecordReader(fileSplit, jobConf, Reporter.NULL);
 
             int headerCount = getHeaderCount(schema);
-            if (headerCount > 0) {
+            //  Files with only a single header row are considered splittable and should only skip the first record on the first split
+            if ((start == 0 && headerCount == 1) || headerCount > 1) {
```

Comment:
When `headerCount > 1` then `verify(start == 0` would be appropriate.

Reply:
Giving this some thought, I think maybe a better formulation is:
```
if (start == 0 && headerCount > 0) {
```

Basically decoupling the knowledge of the special case from the handling here and saying that only splits at the beginning of the file will be considered for header skipping and it's up to the split generator to know when it's safe or not safe to split the file. Thoughts?

Reply:
Sounds good to me.